internal/backends/postgresql/metadata/pool/pool.go

```diff
@@ -134,6 +134,22 @@ func (p *Pool) Get(username, password string) (*pgxpool.Pool, error) {
 	return res, nil
 }
 
+// GetAny returns a random open pool of connections to PostgreSQL, or nil if none are available.
+func (p *Pool) GetAny() *pgxpool.Pool {
+	p.rw.RLock()
+	defer p.rw.RUnlock()
+
+	for _, pool := range p.pools {
```

Comment:
I think this clever approach of getting a first map element might backfire as the same element might be returned multiple times (as in, with a bad distribution). What do you think?

Reply:
Well, more reasons to finish authentication faster 😆 

Seriously, though, the runtime actually makes effort to randomize iterations over maps, so it is as bad as map implementation (which is good enough)